nextest-runner/src/config.rs

```diff
@@ -291,11 +292,76 @@ struct DefaultProfileImpl {
     failure_output: TestOutputDisplay,
     success_output: TestOutputDisplay,
     fail_fast: bool,
-    #[serde(with = "humantime_serde")]
-    slow_timeout: Duration,
+    #[serde(deserialize_with = "require_deserialize_slow_timeout")]
+    slow_timeout: SlowTimeout,
     junit: DefaultJunitImpl,
 }
 
+/// Type for the slow-timeout config key.
+#[derive(Clone, Copy, Debug, Deserialize)]
+#[cfg_attr(test, derive(PartialEq))]
+#[serde(rename_all = "kebab-case")]
+pub struct SlowTimeout {
+    #[serde(with = "humantime_serde")]
+    pub(crate) period: Duration,
+    #[serde(default)]
+    pub(crate) terminate_after: Option<usize>,
```

Comment:
So I'm a little concerned here that terminate_after can be both `Some(0)` and `None`, which are two separate values that just happen to be treated the same. I'm wondering if this can be represented as an `Option<NonZeroUsize>`, or at least with a comment here clearly explaining that `Some(0)` and `None` are the same.

Reply:
i hadn't been aware that `NonZeroUsize` exists as a type. i took your last mentioning of it as a descriptive example :slightly_smiling_face:  as i share your concern about the UX here i'm currently experimenting using `NonZeroUsize`.